test/controllers/diary_entries_controller_test.rb

```diff
@@ -159,7 +159,7 @@ def test_create
       post diary_entries_path(:diary_entry => { :title => "New Title", :body => "This is a new body for the diary entry", :latitude => "1.1",
                                                 :longitude => "2.2", :language_code => "en" })
     end
-    assert_redirected_to :action => :index, :display_name => user.display_name
+    assert_redirected_to diary_entry_path(user, user.diary_entries.last)
     entry = DiaryEntry.last
```

Comment:
It's a bit odd that the same diary entry has two different references next to each other: `user.diary_entries.last` and `DiaryEntry.last`.

Reply:
Yes, I thought that too. I could refactor that as part of this PR, or separately. Which would you prefer? (and do you have any preferences for referring to the entry e.g. `user.diary_entries.last` or `DiaryEntry.last` or something else?)

Reply:
> separately

Won't that be a fixup commit which we don't like here?

> do you have any preferences for referring to the entry

Maybe if we use `assert_difference "DiaryEntry.count"`, we should also use `DiaryEntry.last` (or change `assert_difference`).

Reply:
> Maybe if we use `assert_difference "DiaryEntry.count"`, we should also use `DiaryEntry.last` (or change `assert_difference`).

I've gone for DiaryEntry.last, but rearranged slightly so that we use the `entry` variable for the redirection assertion too. Let me know what you think.